### PR TITLE
deprisma alerting config

### PR DESCRIPTION
**Pull request by serprex**
in prep for encrypting alerting configs

### CODE REVIEW

File: ui/app/alert-config/page.tsx [hunk 1]
```diff
@@ -20,23 +20,21 @@ const ServiceIcon = ({
...
+  />
+);
 
 const AlertConfigPage: React.FC = () => {
   const {
     data: alerts,
     isLoading,
   }: {
-    data: UAlertConfigResponse[];
+    data: { configs: UAlertConfigResponse[] };
```
Comment (heavycrystal) [inline ui/app/alert-config/page.tsx:37]: could replace this with `GetAlertConfigsResponse`

File: protos/route.proto [hunk 2]
```diff
@@ -37,6 +37,31 @@ message CreateCustomSyncResponse {
...
+  int32 id = 1;
+}
+message GetAlertConfigsResponse {
+  repeated AlertConfig configs = 1;
+}
+message PostAlertConfigResponse {
+  int32 id = 3;
+}
+message DeleteAlertConfigResponse {
+}
```
Comment (heavycrystal) [inline protos/route.proto:63]: should contain `ok` at the very least, same for other routes

File: flow/cmd/alerts.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+func (h *FlowRequestHandler) GetAlertConfigs(ctx context.Context, req *protos.GetAlertConfigsRequest) (*protos.GetAlertConfigsResponse, error) {
+	rows, err := h.pool.Query(ctx, "select id, service_type, service_config from alerting_config")
+	if err != nil {
+		return nil, err
+	}
+
+	configs, err := pgx.CollectRows(rows, func(row pgx.CollectableRow) (*protos.AlertConfig, error) {
+		config := &protos.AlertConfig{}
+		err := row.Scan(&config.Id, &config.ServiceType, &config.ServiceConfig)
```
Comment (heavycrystal) [inline flow/cmd/alerts.go:19]: could give https://pkg.go.dev/github.com/jackc/pgx/v5#RowToStructByName a shot

File: protos/route.proto (hunk 2 above)
Comment (serprex) [reply protos/route.proto:63]: seems like errors should rely on grpc error reporting mechanism already there

File: flow/cmd/alerts.go (hunk 3 above)
Comment (serprex) [reply flow/cmd/alerts.go:19]: Don't trust it with protos

**Review by heavycrystal — APPROVED** [review]

**Pull request merged**